File: test_dir/cpp/src/operations.cpp
#include "operations.h"
#include "py_str.h"
#include "pypp_util/math.h"
#include "pypp_util/print.h"
#include <cmath>

void operations_fn() {
    print(PyStr("OPERATIONS RESULTS:"));
    double a = std::pow(2, 3);
    print(PyStr(std::format("float power: {}", a)));
    int b = int_pow(2, 3);
    print(PyStr(std::format("int power: {}", b)));
    int c = 10 % 3;
    print(PyStr(std::format("mod: {}", c)));
}

File: test_dir/python/src/operations.py
from test_dir.python.pypp.math import int_pow


def operations_fn():
    print("OPERATIONS RESULTS:")
    # float power
    a: float = 2**3
    print(f"float power: {a}")
    # int power. Note: for safety, you need to use for integer exponentiation because
    #  C++ does not have safe integer exponentiation.
    b: int = int_pow(2, 3)
    print(f"int power: {b}")
